components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -139,37 +144,73 @@ class Nimbus : NimbusApi {
-
-            // Invoke the callback with the list of active experiments for the consuming app to
-            // process.
-            onExperimentUpdated?.invoke(activeExperiments)
         }
     }
 
     override fun getActiveExperiments(): List<EnrolledExperiment> =
-        if (isInitialized) { nimbus.getActiveExperiments() } else { emptyList() }
+        if (isInitialized) { tryGetActiveExperiments().orEmpty() } else { emptyList() }
 
     override fun getExperimentBranch(experimentId: String): String? =
         if (isInitialized) { nimbus.getExperimentBranch(experimentId) } else { null }
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        // We need to launch this off of the main thread because it involves a network request
+        // within the nimbus-sdk.
+        scope.launch {
+            if (tryUpdateExperiments()) {
+                tryGetActiveExperiments()?.let {
+                    recordExperimentTelemetry(it)
+                    onExperimentUpdated?.invoke(it)
+                }
+            }
+        }
     }
 
     override fun optOut(experimentId: String) {
         if (!isInitialized) return
         nimbus.optOut(experimentId)
     }
 
+    // Wraps the `Nimbus.updateExperiments()` in a try/catch since the underlying nimbus-sdk
+    // makes a network request that might throw an error.  This also wraps the call in
+    // a simple retry mechanism before giving up.
+    // Returns true if the update completed without errors, false otherwise.
+    // Note: a return of "true" doesn't mean that there was any new experiments, just that there
+    // were no errors.
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun tryUpdateExperiments(): Boolean =
+        try {
+            nimbus.updateExperiments()
+            true
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+            false
+        } catch (e: ErrorException.InvalidExperimentResponse) {
+            logger.info("Invalid experiment response: $e")
+            false
+        }
+
+    // Wraps the `Nimbus.getActiveExperiments()` in a try/catch since the underlying nimbus-sdk
+    // call could make a network request that might throw an error.  This also wraps the call in
+    // a simple retry mechanism before giving up.
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun tryGetActiveExperiments(): List<EnrolledExperiment>? =
+        try {
+            nimbus.getActiveExperiments()
+        } catch (e: ErrorException.RequestError) {
+            logger.info("Error fetching experiments from endpoint: $e")
+            null
+        } catch (e: ErrorException.InvalidExperimentResponse) {
+            logger.info("Invalid experiment response: $e")
+            null
```

Comment:
At this point, does `getActiveExperiments()` throw any network errors?

If it does, we should consider not doing a network call in a simple `get` method.

Reply:
It shouldn't do any network requests, but the version that we are currently consuming from appservices _could_, but that should change once we get a new release of nimbus-sdk.